lib/Backends/CPU/libjit/libjit.cpp

```diff
@@ -1410,4 +1410,125 @@ libjit_dump_tensor(uint8_t *tensor, size_t *tensorDim, size_t numDimsTensor,
     break;
   }
 }
+
+static void find_min_max_f(float *tensor, size_t size, float &min, float &max) {
+  min = tensor[0];
+  max = tensor[0];
+
+  for (size_t i = 1; i < size; ++i) {
+    float tensorVal = tensor[i];
+    if (tensorVal < min)
+      min = tensorVal;
+
+    if (tensorVal > max)
+      max = tensorVal;
+  }
+}
+
+static int check_all_zeros(float *arrayToCheck, size_t size) {
+  int are_all_zero = 1;
+  for (int i = 0; i < size; ++i) {
+    if (arrayToCheck[i] != 0) {
+      return 0;
+    }
+  }
+  return 1;
+}
+
+/// Gen a bin number to insert \p value into the histogram which has \p nBins
+/// with \p minValue and binWidth in histogram.
+static size_t getBin(size_t nBins, float binWidth, float minValue,
+                     float value) {
+  size_t result =
+      binWidth == 0
+          ? 0
+          : MIN(static_cast<size_t>((value - minValue) / binWidth), nBins - 1);
+  return result;
+}
+
+// code ported from Profile.cpp: generateTensorHistogram
+__attribute__((noinline)) void
```

Comment:
Nit: `__attribute__((noinline))` is not guaranteed to have any effect at the moment (see https://github.com/pytorch/glow/blob/master/lib/Backends/CPU/Pipeline.cpp#L79-L83 for more details)

Reply:
Thanks for the link. I was following the definitions of other libjit functions in this file. Should I remove the attribute, or keep it for consistency with other functions?

Reply:
@ayermolo You can keep it for consistency with other functions.